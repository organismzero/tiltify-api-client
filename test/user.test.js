const TiltifyClient = require('../index')

describe('user tests', () => {
  beforeAll(() => {
    this.client = new TiltifyClient('','').setKey(process.env.TILTIFY_ACCESS_TOKEN)
  })

  test.skip('correct Tiltify user self returned', done => {
    function callback (data) {
      expect(data).toMatchSnapshot()
      done()
    }
    this.client.User.self(callback)
  })

  test('correct Tiltify user returned', done => {
    function callback (data) {
      expect(data).toMatchSnapshot()
      done()
    }
    this.client.User.get('3935', callback)
  })

  test('correct Tiltify user campaigns returned', done => {
    function callback (data) {
      expect(data).toMatchSnapshot()
      done()
    }
    this.client.User.getSelfCampaigns('3935', callback)
  })

  test('correct Tiltify user teams returned', done => {
    function callback (data) {
      expect(data).toMatchSnapshot()
      done()
    }
    this.client.User.getTeams('3935', callback)
  })

})
