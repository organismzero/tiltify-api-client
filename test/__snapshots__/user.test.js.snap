// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user tests correct Tiltify user campaigns returned 1`] = `
Array [
  Object {
    "amount_raised": Object {
      "currency": "USD",
      "value": "36.00",
    },
    "avatar": Object {
      "alt": "alt",
      "height": 64,
      "src": "https://assets.tiltify.com/uploads/cause/avatar/65/direct-relief-avatar-1588270117181.png",
      "width": 64,
    },
    "cause_id": "17ab0696-167a-4e13-85b7-4704aed62e30",
    "description": "Hi! I'm Ender3989 and I like to play games. I also would love if you'd donate money to Direct Relief!",
    "fundraising_event_id": null,
    "goal": Object {
      "currency": "USD",
      "value": "100.0",
    },
    "has_schedule": false,
    "id": "b827d321-e8b7-4de4-bee8-199abd26165a",
    "inserted_at": "2016-10-18T22:19:42Z",
    "legacy_id": 4814,
    "livestream": Object {
      "channel": "ender3989",
      "type": "twitch",
    },
    "name": "Playing Games for Charity!",
    "original_goal": Object {
      "currency": "USD",
      "value": "100.0",
    },
    "published_at": "2016-10-29T21:34:09Z",
    "retired_at": "2018-11-28T18:08:20Z",
    "slug": "playing-games-for-charity",
    "status": "retired",
    "supporting_type": "private",
    "total_amount_raised": Object {
      "currency": "USD",
      "value": "36.00",
    },
    "updated_at": "2022-06-14T17:22:12Z",
    "url": "/@ender3989/playing-games-for-charity",
    "user": Object {
      "avatar": Object {
        "alt": "alt",
        "height": 200,
        "src": "https://assets.tiltify.com/assets/default-avatar.png",
        "width": 200,
      },
      "description": "",
      "id": "565494e9-378d-46e2-a24d-afbfe3e63c8d",
      "legacy_id": 3935,
      "slug": "ender3989",
      "social": Object {
        "discord": null,
        "facebook": null,
        "instagram": null,
        "snapchat": null,
        "tiktok": null,
        "twitch": null,
        "twitter": null,
        "website": null,
        "youtube": null,
      },
      "total_amount_raised": Object {
        "currency": "USD",
        "value": "36.00",
      },
      "url": "/@ender3989",
      "username": "ender3989",
    },
    "user_id": "565494e9-378d-46e2-a24d-afbfe3e63c8d",
  },
  Object {
    "amount_raised": Object {
      "currency": "USD",
      "value": "0.00",
    },
    "avatar": Object {
      "alt": "alt",
      "height": 64,
      "src": "https://assets.tiltify.com/uploads/cause/avatar/65/direct-relief-avatar-1588270117181.png",
      "width": 64,
    },
    "cause_id": "17ab0696-167a-4e13-85b7-4704aed62e30",
    "description": null,
    "fundraising_event_id": null,
    "goal": Object {
      "currency": "USD",
      "value": "100.0",
    },
    "has_schedule": true,
    "id": "e6d44e87-1290-430f-9a25-2efe79327b86",
    "inserted_at": "2018-10-07T17:06:58Z",
    "legacy_id": 19414,
    "livestream": null,
    "name": "Danny's Test Campaign",
    "original_goal": Object {
      "currency": "USD",
      "value": "100.0",
    },
    "published_at": "2019-05-01T00:23:59Z",
    "retired_at": "2019-12-08T01:00:50Z",
    "slug": "dannys-test-campaign",
    "status": "retired",
    "supporting_type": "private",
    "total_amount_raised": Object {
      "currency": "USD",
      "value": "0.00",
    },
    "updated_at": "2022-06-14T17:22:38Z",
    "url": "/@ender3989/dannys-test-campaign",
    "user": Object {
      "avatar": Object {
        "alt": "alt",
        "height": 200,
        "src": "https://assets.tiltify.com/assets/default-avatar.png",
        "width": 200,
      },
      "description": "",
      "id": "565494e9-378d-46e2-a24d-afbfe3e63c8d",
      "legacy_id": 3935,
      "slug": "ender3989",
      "social": Object {
        "discord": null,
        "facebook": null,
        "instagram": null,
        "snapchat": null,
        "tiktok": null,
        "twitch": null,
        "twitter": null,
        "website": null,
        "youtube": null,
      },
      "total_amount_raised": Object {
        "currency": "USD",
        "value": "36.00",
      },
      "url": "/@ender3989",
      "username": "ender3989",
    },
    "user_id": "565494e9-378d-46e2-a24d-afbfe3e63c8d",
  },
  Object {
    "amount_raised": Object {
      "currency": "USD",
      "value": "0.00",
    },
    "avatar": Object {
      "alt": "alt",
      "height": 64,
      "src": "https://assets.tiltify.com/uploads/cause/avatar/35/blob-69ddc696-2034-4d43-b6e0-dc7049f035fe.png",
      "width": 64,
    },
    "cause_id": "400f5687-6017-4d1a-a4d9-7c9166b984c2",
    "description": "test",
    "fundraising_event_id": null,
    "goal": Object {
      "currency": "USD",
      "value": "100",
    },
    "has_schedule": true,
    "id": "76ae157c-1ecb-4774-982b-380e87dddc97",
    "inserted_at": "2021-05-11T20:15:33Z",
    "legacy_id": 116627,
    "livestream": null,
    "name": "test",
    "original_goal": Object {
      "currency": "USD",
      "value": "100",
    },
    "published_at": "2021-05-11T20:17:15Z",
    "retired_at": "2021-07-11T01:02:31Z",
    "slug": "test",
    "status": "retired",
    "supporting_type": "private",
    "total_amount_raised": Object {
      "currency": "USD",
      "value": "0.00",
    },
    "updated_at": "2022-06-14T17:19:25Z",
    "url": "/@ender3989/test",
    "user": Object {
      "avatar": Object {
        "alt": "alt",
        "height": 200,
        "src": "https://assets.tiltify.com/assets/default-avatar.png",
        "width": 200,
      },
      "description": "",
      "id": "565494e9-378d-46e2-a24d-afbfe3e63c8d",
      "legacy_id": 3935,
      "slug": "ender3989",
      "social": Object {
        "discord": null,
        "facebook": null,
        "instagram": null,
        "snapchat": null,
        "tiktok": null,
        "twitch": null,
        "twitter": null,
        "website": null,
        "youtube": null,
      },
      "total_amount_raised": Object {
        "currency": "USD",
        "value": "36.00",
      },
      "url": "/@ender3989",
      "username": "ender3989",
    },
    "user_id": "565494e9-378d-46e2-a24d-afbfe3e63c8d",
  },
]
`;

exports[`user tests correct Tiltify user returned 1`] = `
Array [
  Object {
    "avatar": Object {
      "alt": "alt",
      "height": 200,
      "src": "https://assets.tiltify.com/assets/default-avatar.png",
      "width": 200,
    },
    "description": "",
    "id": "565494e9-378d-46e2-a24d-afbfe3e63c8d",
    "legacy_id": 3935,
    "slug": "ender3989",
    "social": Object {
      "discord": null,
      "facebook": null,
      "instagram": null,
      "snapchat": null,
      "tiktok": null,
      "twitch": null,
      "twitter": null,
      "website": null,
      "youtube": null,
    },
    "total_amount_raised": Object {
      "currency": "USD",
      "value": "36.00",
    },
    "url": "/@ender3989",
    "username": "ender3989",
  },
]
`;

exports[`user tests correct Tiltify user teams returned 1`] = `Array []`;
